core/jvm/src/main/java/zio/internal/Signal.java

```diff
@@ -0,0 +1,151 @@
+        } catch (ClassNotFoundException e) {
+            return null;
+        }
+    }
+
+    private static MethodHandle initSignalConstructorMethodHandle(
+            MethodHandles.Lookup lookup,
+            Class<?> signalClass) {
+        final MethodType signalConstructorMethodType = MethodType.methodType(void.class, String.class);
+        try {
+            return lookup.findConstructor(signalClass, signalConstructorMethodType);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private static MethodHandle initHandleStaticMethodHandle(
+            MethodHandles.Lookup lookup,
+            Class<?> signalClass,
+            Class<?> signalHandlerClass) {
+        try {
+            final MethodType handleStaticMethodType = MethodType.methodType(signalHandlerClass, signalClass, signalHandlerClass);
+            return lookup.findStatic(signalClass, "handle", handleStaticMethodType);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private abstract static class SignalHandler {
+        abstract void handle(String signal, Consumer<Object> handler);
+
+        private final static class SunMiscSignalHandler extends SignalHandler {
+            Class<?> signalHandlerClass;
+            MethodHandle constuctorMethodHandle;
+            MethodHandle staticMethodHandle;
+
+            private SunMiscSignalHandler(
+                    Class<?> signalHandlerClass,
+                    MethodHandle constuctorMethodHandle,
+                    MethodHandle staticMethodHandle
+            ) {
+                assertNotNull(signalHandlerClass, "signalHandlerClass");
+                assertNotNull(constuctorMethodHandle, "constuctorMethodHandle");
+                assertNotNull(staticMethodHandle, "staticMethodHandle");
+
+                this.signalHandlerClass = signalHandlerClass;
+                this.constuctorMethodHandle = constuctorMethodHandle;
+                this.staticMethodHandle = staticMethodHandle;
+            }
+
+            private static void assertNotNull(Object obj, String message) {
+                if (obj == null) {
+                    throw new IllegalArgumentException(message + " cannot be null");
+                }
+            }
+
+            @Override
+            void handle(String signal, Consumer<Object> handler) {
+                final InvocationHandler invocationHandler = initInvocationHandler(handler);
+                final Object proxy = Proxy.newProxyInstance(
+                        Signal.class.getClassLoader(),
+                        new Class<?>[]{signalHandlerClass},
+                        invocationHandler
+                );
+                try {
+                    final Object s = constuctorMethodHandle.invoke(signal);
+                    staticMethodHandle.invoke(s, proxy);
+                } catch (Throwable err) {
+                    // Gracefully degrade to a no-op logging the exception.
+                }
```

Comment:
Is it safe to catch all `Throwable`?

Also, is there a reason to use `Throwable` here and `Exception` above?

Reply:
It's mostly because the `invoke` method can throw any Throwable (depending on the underlying method) whereas the method above only throw exceptions. I think it's relatively safe to catch all Throwable's here because I don't think we want to crash the application in the very unlikely edge-case that we failed to handle a signal